cwms-data-api/src/main/java/cwms/cda/api/TextTimeSeriesController.java

```diff
@@ -0,0 +1,357 @@
+
+import com.codahale.metrics.MetricRegistry;
+import com.codahale.metrics.Timer;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import cwms.cda.api.errors.CdaError;
+import cwms.cda.data.dao.texttimeseries.TimeSeriesTextDao;
+import cwms.cda.data.dao.texttimeseries.TimeSeriesTextMode;
+import cwms.cda.data.dto.texttimeseries.TextTimeSeries;
+import cwms.cda.formatters.ContentType;
+import cwms.cda.formatters.Formats;
+import cwms.cda.formatters.json.JsonV2;
+import cwms.cda.helpers.DateUtils;
+import io.javalin.apibuilder.CrudHandler;
+import io.javalin.core.util.Header;
+import io.javalin.http.Context;
+import io.javalin.plugin.openapi.annotations.HttpMethod;
+import io.javalin.plugin.openapi.annotations.OpenApi;
+import io.javalin.plugin.openapi.annotations.OpenApiContent;
+import io.javalin.plugin.openapi.annotations.OpenApiParam;
+import io.javalin.plugin.openapi.annotations.OpenApiRequestBody;
+import io.javalin.plugin.openapi.annotations.OpenApiResponse;
+import java.time.ZonedDateTime;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+import javax.servlet.http.HttpServletResponse;
+import org.jetbrains.annotations.NotNull;
+import org.jetbrains.annotations.Nullable;
+import org.jooq.DSLContext;
+
+
+public class TextTimeSeriesController implements CrudHandler {
+    private static final Logger logger = Logger.getLogger(TextTimeSeriesController.class.getName());
+    private static final String TAG = "Text-TimeSeries";
+    public static final String TS_ID = "ts-id";
+    public static final String REPLACE_ALL = "replace-all";
+    private final MetricRegistry metrics;
+
+
+
+    public TextTimeSeriesController(MetricRegistry metrics) {
+        this.metrics = metrics;
+    }
+
+    @NotNull
+    protected TimeSeriesTextDao getDao(DSLContext dsl) {
+        return new TimeSeriesTextDao(dsl);
+    }
+
+
+    private Timer.Context markAndTime(String subject) {
+        return Controllers.markAndTime(metrics, getClass().getName(), subject);
+    }
+
+
+    @OpenApi(
+            queryParams = {
+                    @OpenApiParam(name = OFFICE, required = true, description = "Specifies the owning office of "
+                            + "the Text TimeSeries whose data is to be included in the response."),
+                    @OpenApiParam(name = TS_ID, required = true, description = "Specifies the ts-id of the "
+                            + "text timeseries"),
+                    @OpenApiParam(name = TIMEZONE,  description = "Specifies "
+                            + "the time zone of the values of the begin and end fields (unless "
+                            + "otherwise specified). If this field is not specified, "
+                            + "the default time zone of UTC shall be used."),
+                    @OpenApiParam(name = START, required = true, description = "The start of the time window"),
```

Comment:
numerical time series uses `BEGIN`

Reply:
also, is the default time window of the last 24 hours applicable here (to match numerical time series)? or should the `END` parameter also be required?